pkg/serverconf/file_ini.go

```diff
@@ -0,0 +1,106 @@
+package serverconf
+
+import (
+	"path/filepath"
+	"strconv"
+
+	"gopkg.in/ini.v1"
+)
+
+type inicfg struct {
+	file         *ini.File
+	murmurCompat bool
+}
+
+func newinicfg(path string) (*inicfg, error) {
+	file, err := ini.LoadSources(ini.LoadOptions{AllowBooleanKeys: true, UnescapeValueDoubleQuotes: true}, path)
+	if err != nil {
+		return nil, err
+	}
+	file.BlockMode = false // read only, avoid locking
+	return &inicfg{file, filepath.Base(path) == "murmur.ini"}, nil
+}
+
+func (f *inicfg) GlobalMap() map[string]string {
+	if !f.murmurCompat {
+		return f.file.Section("").KeysHash()
+	} else {
+		return TranslateMurmur(f.file.Section("").KeysHash())
```

Comment:
i think it would be preferable if the ini file had as many common fields with the murmur config as possible.  Can we rename all config values appropriately (e.g. Address -> host)

Reply:
There are a few reasons I didn't do this.
1) Backwards compatibility with previous grumble freezes. The current way supports both the previous/current grumble key names and the previous murmur key names. (I don't know if this matters, and I actually accidentally broke it by renaming MaxChannelUsers to MaxUsersPerChannel. Oops! Fixing that when I have time to write a patch.)
2) I want a TranslateMurmur step and a list of supported murmur keys anyway to print messages regarding common unsupported configuration. (It also strips unsupported keys as to not pollute the freeze files, but that will be unnecessary later, see below.)
3) Subjective: My interpretation of #21 is that the murmur support should be a non-default _personality_. (I chose to trigger it if the config file is named murmur.ini, but other solutions are possible, too.) This means that the normal configuration isn't dependent on murmur, so there's no guarantee everything works the same way in normal mode. (However, the murmur mode instead promises compatibility -- two sides of the same coin.) This means there is no need to rename things like Address either (it's had that name for 7 years). Rather, murmur configuration should be converted.

Because of the freeze persistence, grumble _still_ wouldn't behave like murmur when you try to reconfigure murmur.ini to defaults. I didn't have time to fix that just yet, but I want the config to not persist in the freeze (overriding defaults, but not config set through future RPC).

This obviously does not preclude renaming the new keys closer to their murmur counterparts (like TLS* to SSL*, *Path to *File etc.), and fixing the persisting configs are on my to-do list. Even if you're not convinced about the field names, I'd still like to hear what you think about the translation/filter step.